### PR TITLE
Rename vjoinsplit to vJoinSplit

### PR DESCRIPTION
**Pull request by rex4539**
Closes https://github.com/zcash/zcash/issues/3188

### CODE REVIEW

File: qa/rpc-tests/paymentdisclosure.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def run_test (self):
         txid = wait_and_assert_operationid_status(self.nodes[0], myopid)
 
         # Check the tx has joinsplits
-        assert( len(self.nodes[0].getrawtransaction("" + txid, 1)["vjoinsplit"]) > 0 )
+        assert( len(self.nodes[0].getrawtransaction("" + txid, 1)["vJoinSplit"]) > 0 )
```
Comment (daira) [inline qa/rpc-tests/paymentdisclosure.py]: Hmm, I hadn't realized this was used in `getrawtransaction` output. Changing that is strictly speaking a compatibility break. I think the original intent of #3188 was just to change it in the C++ code.

Comment (rex4539) [reply qa/rpc-tests/paymentdisclosure.py]: Reverted this instance. Let me know if there are any other compatibility breaks.

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -105,7 +105,7 @@ void WalletTxToJSON(const CWalletTx& wtx, UniValue& entry)
     BOOST_FOREACH(const PAIRTYPE(string,string)& item, wtx.mapValue)
         entry.push_back(Pair(item.first, item.second));
 
-    entry.push_back(Pair("vjoinsplit", TxJoinSplitToJSON(wtx)));
+    entry.push_back(Pair("vJoinSplit", TxJoinSplitToJSON(wtx)));
```
Comment (daira) [inline src/wallet/rpcwallet.cpp]: This shouldn't be changed.

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -1741,7 +1741,7 @@ UniValue gettransaction(const UniValue& params, bool fHelp)
             "    }\n"
             "    ,...\n"
             "  ],\n"
-            "  \"vjoinsplit\" : [\n"
+            "  \"vJoinSplit\" : [\n"
```
Comment (daira) [inline src/wallet/rpcwallet.cpp]: This shouldn't be changed.

File: src/rpc/rawtransaction.cpp [hunk 4]
```diff
@@ -212,8 +212,8 @@ void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry)
     }
     entry.push_back(Pair("vout", vout));
 
-    UniValue vjoinsplit = TxJoinSplitToJSON(tx);
-    entry.push_back(Pair("vjoinsplit", vjoinsplit));
+    UniValue vJoinSplit = TxJoinSplitToJSON(tx);
+    entry.push_back(Pair("vJoinSplit", vJoinSplit));
```
Comment (daira) [inline src/rpc/rawtransaction.cpp]: This shouldn't be changed.

**Comment by rex4539** [comment]
Addressed all comments.

File: src/wallet/rpcwallet.cpp [hunk 5]
```diff
@@ -4235,7 +4235,7 @@ UniValue z_shieldcoinbase(const UniValue& params, bool fHelp)
     CMutableTransaction contextualTx = CreateNewContextualCMutableTransaction(
         Params().GetConsensus(), nextBlockHeight);
     if (contextualTx.nVersion == 1) {
-        contextualTx.nVersion = 2; // Tx format should support vjoinsplits 
+        contextualTx.nVersion = 2; // Tx format should support vJoinSplits
```
Comment (daira) [inline src/wallet/rpcwallet.cpp]: Nitpick: this should be `vJoinSplit`.

Comment (rex4539) [reply src/wallet/rpcwallet.cpp]: Fixed.

**Comment by Eirik0** [comment]
@zkbot r+

**Pull request merged**